src/biome/text/model_instances/model.py

```diff
@@ -0,0 +1,229 @@
+        -------
+            The configured ``allennlp.models.BaseModelInstance``
+        """
+        return self._model
+
+    @property
+    def allennlp_config(self) -> dict:
+        """
+        A representation of reader and model in a properties defined way
+        as allennlp does
+
+        Returns
+        -------
+            The configuration dictionary
+        """
+        return self.__config.copy()
+
+    def predict(self, **inputs) -> dict:
+        return self.predict_json(inputs)
+
+    @classmethod
+    def load(cls, path: str, **kwargs) -> "BaseModelInstance":
+        name = cls.__to_snake_case(cls.__name__)
+        Predictor.register(name, exist_ok=True)(cls)
+        archive = load_archive(path, **kwargs)
+        return cls.from_archive(archive, predictor_name=name)
+
+    @overrides
+    def _json_to_instance(self, json_dict: JsonDict) -> Instance:
+        return self.pipeline.text_to_instance_with_data_filter(json_dict)
+
+    @overrides
+    def predict_json(self, inputs: JsonDict) -> JsonDict:
+        from allennlp.common.util import sanitize
+
+        instance = self._json_to_instance(inputs)
+        output = self.architecture.forward_on_instance(instance)
+        return sanitize(output)
+
+    @staticmethod
+    def __to_snake_case(name):
+        """
+        A helper method for convert a CamelCase name into a snake_case name
+
+        Parameters
+        ----------
+        name
+            The original name
+
+        Returns
+        -------
+            The corresponding snake_case name
+
+        """
+        s1 = re.sub(r"(.)([A-Z][a-z]+)", r"\1_\2", name)
+        return re.sub(r"([a-z0-9])([A-Z])", r"\1_\2", s1).lower()
+
+    @staticmethod
+    def yaml_to_dict(filepath: str):
+        with open(filepath) as yaml_content:
+            config = yaml.safe_load(yaml_content)
+        return config
+
+    @classmethod
+    def from_config(cls, path: str) -> "BaseModelInstance":
+        """
+        Read a ``BaseModelInstance`` subclass instance by reading a configuration file
+
+        Parameters
+        ----------
+        path
+            The configuration file path
+
+        Returns
+        -------
+            An instance of ``BaseModelInstance`` with no architecture, since the internal
+            ``allennlp.models.Model`` needs a Vocabulary for the initialization
+
+        """
+        data = cls.yaml_to_dict(path)
+        # backward compatibility
+        if data.get("topology"):
+            data = data["topology"]
+        main_class = cast(
+            Type[BaseModelInstance], Predictor.by_name(data.get("type", cls.__name__))
+        )
+        name = cls.__to_snake_case(main_class.__name__)
+
+        model = main_class(
+            model=None,
+            reader=cast(
+                DataSourceReader,
+                DatasetReader.from_params(
+                    Params({**data["pipeline"], "type": name}.copy())
+                ),
+            ),
+        )
```

Comment:
To me this looks super obscure ... maybe some more comments :) or more helper functions with meaningful names? Why do we use a `Predictor` to get the name? Why do we use type `BaseModelInstance` for all classes derived from `BaseModelInstance`? What if the registered classes do not follow the snake case practice?

If a model cannot be initiated here, maybe it would be sufficient to just hold the model.__config on init, and once we have executed a training, it will get Model and DatasetReader instances.